### DISCUSSION TITLE
HyperGraphData with two levels data object

### DISCUSSION THREAD

**Discussion by CaiYitao** [pyg-team/pytorch_geometric#8951] [Q&A]
I am working on a chemical reaction network problem.  the chemical reaction network is a hypergraph, each node is mol(reactant or product) , each hyperedge indicate chemical reaction rule. I am thinking of constructing the hypergraph data set as HyperGraphData(x=[Data(x=atom_attr, edge_index = edge_index), ..., Data()],  hyperedge_index=he_idx, hyperedge_attr= [[Data(x=rule_attr,..), Data(), Data()],...,])    in which hypergraph vertices are Data object for molecules.,  hyperedge attributes are a list of n by 3 array of rule data object. Can we do something like this?   In Hypergraph Data object it states that the x should be tensor.

**Comment by wsad1** [comment]
The [HyperGraphData](https://github.com/pyg-team/pytorch_geometric/blob/25b2f208e671eeec285bfafa2e246ea0a234b312/torch_geometric/data/hypergraph_data.py#L15) object support `edge_attr`, you could use it to set the attribute of edges.

**Reply by CaiYitao** [reply]
Yeah, I am planning to add hyperedge_attr as [[Data(x=rule_attr,..), Data(), Data()],...,], it is an array of Data object  just like hypergraph node attribute as a list of Data object. is this doable?
